From: Dynamic Devices Ltd <info@dynamicdevices.co.uk>
Date: Wed, 13 Sep 2025 00:00:00 +0000
Subject: [PATCH] Fix crash on Intel 7265 WiFi by setting default cmd version

This patch fixes a crash that occurs on Intel 7265 WiFi devices by
setting a default command version instead of using IWL_FW_CMD_VER_UNKNOWN.

Upstream-Status: Inappropriate [workaround for specific hardware issue]

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/fw.c b/drivers/net/wireless/intel/iwlwifi/mvm/fw.c
index e7f5978ef2d719..ca7fcae6995ee3 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/fw.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/fw.c
@@ -895,8 +895,8 @@ int iwl_mvm_sar_select_profile(struct iwl_mvm *mvm, int prof_a, int prof_b)
 	int ret;
 	u16 len = 0;
 	u32 n_subbands;
-	u8 cmd_ver = iwl_fw_lookup_cmd_ver(mvm->fw, cmd_id,
-					   IWL_FW_CMD_VER_UNKNOWN);
+	u8 cmd_ver = iwl_fw_lookup_cmd_ver(mvm->fw, cmd_id, 3);
+
 	if (cmd_ver >= 7) {
 		len = sizeof(cmd.v7);
 		n_subbands = IWL_NUM_SUB_BANDS_V2;
